### PR TITLE
docs: Updated the Snapshots Documents reg. Snapshot support for volumes with more than one replica

### PR DESCRIPTION
**Pull request by balaharish7**

### CODE REVIEW

File: docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot-restore.md [hunk 1]
```diff
@@ -15,63 +15,67 @@ Volume restore from an existing snapshot will create an exact replica of a stora
 To begin, you will need to create a StorageClass that defines the properties of the snapshot to be restored. Refer to [StorageClass Parameters](../../replicated-pv-mayastor/rs-configuration.md#storage-class-parameters) for more details. Use the following command to create the StorageClass:
 
 :::info
-thin: "true" and repl: "1" is the only supported combination.
+If you want to create a volume restore from snapshots, thin: "true" is required.
```
Comment (Abhinandan-Purkait) [inline docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot-restore.md:18]: For creation of volume restore from snapshots, storage class parameter, `thin: "true"` is required.

File: docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ keywords:
  - Volume snapshots
 description: This guide explains about the Volume snapshots feature.
 ---
-**Volume snapshots** are copies of a persistent volume at a specific point in time. They can be used to restore a volume to a previous state or create a new volume. Replicated PV Mayastor provides support for industry standard copy-on-write (COW) snapshots, which is a popular methodology for taking snapshots by keeping track of only those blocks that have changed.
+**Volume snapshots** are copies of a persistent volume at a specific point in time. They can restore a volume to a previous state or create a new volume. Replicated PV Mayastor provides support for industry-standard copy-on-write (COW) snapshots, which is a popular methodology for taking snapshots by keeping track of only those blocks that have changed.
```
Comment (Abhinandan-Purkait) [inline docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot.md:8]: industry-standard --> industry standard

File: docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot.md [hunk 3]
```diff
@@ -69,14 +66,18 @@ kubectl get pvc
...
 :::note
 Copy the PVC name, for example, `ms-volume-claim`.
 :::
 
-4. (Optional) Create an application by following [these](../rs-deployment.md) steps.
+4. Create an application by following [these](../rs-deployment.md) steps.
+
+:::tip
+This step is optional.
+:::
```
Comment (Abhinandan-Purkait) [inline docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot.md:80]: Let's not mention that this is optional. Volume will be consumed by application at some point that's what they are meant for.

**Review by Abhinandan-Purkait — APPROVED** [review]

File: docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot-restore.md (hunk 1 above)
Comment (tiagolobocastro) [reply docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot-restore.md:18]: I would add a bit more, something like this:

```suggestion
As it stands today, restores are thin-provisioned volumes created from a snapshot, and as such the storage class for the restored volume must also specify `thin: "true"`.
```

File: docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot.md (hunk 2 above)
Comment (tiagolobocastro) [inline docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot.md:8]:
```suggestion
**Volume snapshots** are copies of a persistent volume at a specific point in time. They can restore a volume to a previous state or create a new volume. Replicated PV Mayastor provides support for industry standard copy-on-write (COW) snapshots, which is a popular methodology for taking snapshots by keeping track of changed blocks.
```

**Review by tiagolobocastro — APPROVED** [review]

File: docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot.md (hunk 2 above)
Comment (balaharish7) [reply docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot.md:8]: I added a Hyphen as these 2 words together form an adjective phrase modifying a noun. :)

File: docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot.md [hunk 4]
```diff
@@ -24,9 +24,13 @@ Currently, Replicated PV Mayastor supports the following operations related to v
 2. Listing available snapshots for a PVC
 3. Deleting a snapshot for a PVC
 
+:::info
+Unlike volume replicas, snapshots cannot be rebuild on an event of a node failure.
```
Comment (avishnu) [inline docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot.md:28]: rebuilt*

File: docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot-restore.md [hunk 5]
```diff
@@ -8,14 +8,56 @@ description: This guide explains about the Snapshot Restore feature.
 
 Volume restore from an existing snapshot will create an exact replica of a storage volume captured at a specific point in time. They serve as an essential tool for data protection, recovery, and efficient management in Kubernetes environments. This article provides a step-by-step guide on how to create a volume restore.
 
+:::info
+Snapshots can be restored if the number of available snapshots is less than or equal to the replicas.
```
Comment (avishnu) [inline docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot-restore.md:12]: This statement needs to be explained better. It is vague and confusing.

Comment (avishnu) [reply docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/advanced-operations/snapshot-restore.md:12]: Maybe something like the below:
A volume snapshot transcends into taking snapshots on each of the volume's replicas for consistency. For instance, a snapshot on a source volume with three replicas (repl=3) results in snapshot-ting all three replicas of the source volume.
Likewise, when a restore is invoked i.e. a new volume is provisioned by specifying the source volume's snapshot as the source, the replica count for the new volume needs to be less than or equal to the available replica snapshot count of the corresponding source volume.

**Review by avishnu — APPROVED** [review]

**Pull request merged**